src/flb_sds.c

```diff
@@ -138,6 +138,11 @@ flb_sds_t flb_sds_cat(flb_sds_t s, const char *str, int len)
     return s;
 }
 
+flb_sds_t flb_sds_cat_auto(flb_sds_t s, const char *str)
+{
+    return flb_sds_cat(s, str, FLB_SDS_HEADER(s)->len);
```

Comment:
I think the given length is not right, the concatenation length aims to specify how long is the string your are appending, not the length of the original buffer you are appending to, it needs to be something like:

```c
int len;

len = strlen(str);
return flb_sds_cat(s, str, len);
```

Reply:
Thanks and DONE!